tests/adam_test.cpp

```diff
@@ -418,3 +418,51 @@ TEST_CASE("PadamLogisticRegressionTest", "[AdamTest]")
       coordinates);
   REQUIRE(testAcc == Approx(100.0).epsilon(0.006)); // 0.6% error tolerance.
 }
+
+TEST_CASE("SimpleAdamWTestFunction", "[AdamTest]")
```

Comment:
Should we add this one to the ``adamr_test.cpp`` file?

Reply:
Adamr is for Test groups with Warm restarts , whereas AdamW is a Adam Update . I dont think we should add this to adamr.

Reply:
Thanks for the clarification, makes sense.